### PR TITLE
Fixes metrics for log bytes sent

### PR DESCRIPTION
**Pull request by KrisKennawayDD**
### What does this PR do?

When the telemetry package was introduced in 09957f78be52364182cd6dbebc7364d781c2b334 versions of the logs `(encoded_)bytes_sent` metrics were added to `pkg/logs/client/tcp/destination.go` but not `pkg/logs/client/http/destination.go`

This means that currently the only way to obtain log telemetry that correctly includes http transport is via the legacy expvar metrics.

I also noticed that the tcp version uses the encoded payload size for both the encoded and unencoded byte sizes.  This was confusing to me so I've changed this to explicitly populate UnencodedSize and use it on the metrics.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/logs/client/tcp/destination.go [hunk 1]
```diff
@@ -101,8 +101,8 @@ func (d *Destination) sendAndRetry(payload *message.Payload, output chan *messag
 
 		metrics.LogsSent.Add(1)
 		metrics.TlmLogsSent.Inc()
-		metrics.BytesSent.Add(int64(len(payload.Encoded)))
-		metrics.TlmBytesSent.Add(float64(len(payload.Encoded)))
+		metrics.BytesSent.Add(int64(payload.UnencodedSize))
```
Comment (gh123man) [inline pkg/logs/client/tcp/destination.go:104]: the `UnencodedSize` [is not set for TCP payloads](https://github.com/DataDog/datadog-agent/blob/main/pkg/logs/sender/stream_strategy.go#L37) as TCP uses the stream strategy which does no encoding. 

This is a bit of an implementation detail but in order for this to work you could either set the `UnencodedSize` to `len(payload.Encoded)` at the line I linked above - or revert this back to use `len(payload.Encoded)` here.

Comment (KrisKennawayDD) [reply pkg/logs/client/tcp/destination.go:104]: Ah, thank you for catching - I've changed to set UnencodedSize explicitly, which I think makes the intent more obviously correct for the metrics.

**Review by gh123man — APPROVED** [review]
LGTM

**Review by apigirl — COMMENTED** [review]
release note is fine for editorial review

**Comment by gh123man** [comment]
@KrisKennawayDD I've added a milestone and team_label for you. For some reason I can't re-run the jobs. but otherwise this is safe to merge so I'll merge it now.

**Comment by KrisKennawayDD** [comment]
Thank you Brian!

**Pull request merged**